cmd/icingadb/main.go

```diff
@@ -279,6 +288,21 @@ func run() int {
 							return rt.Sync(synctx, v1.StateFactories, runtimeStateUpdateStreams, true)
 						})
 
+						g.Go(func() error {
+							// Wait for config and state sync to avoid putting additional pressure on the database.
+							configInitSync.Wait()
+							stateInitSync.Wait()
+
+							select {
+							case <-synctx.Done():
+								return synctx.Err()
+							default:
```

Comment:
Could be made a simple if. `Err() != nil` iff `Done()` is closed: https://pkg.go.dev/context#Context

Reply:
Done.